src/RetrySettings.php

```diff
@@ -32,44 +32,258 @@
 namespace Google\GAX;
 
 /**
- * Holds the parameters for retry and timeout logic with exponential backoff. Actual
- * implementation of the logic is elsewhere.
+ * The RetrySettings class is used to configure retrying and timeouts for RPCs.
+ * This class can be passed as an optional parameter to RPC methods, or as part
+ * of an optional array in the constructor of a client object. In addition,
+ * many RPCs and API clients accept a PHP array in place of a RetrySettings
+ * object. This can be used to change particular retry parameters without
+ * needing to construct a complete RetrySettings object.
  *
- * The intent of these settings is to be used with a call to a remote server, which
- * could either fail (and return an error code) or not respond (and cause a timeout).
- * When there is a failure or timeout, the logic should keep trying until the total
- * timeout has passed.
+ * Constructing a RetrySettings object
+ * -----------------------------------
+ *
+ * See the RetrySettings constructor for documentation about parameters that
+ * can be passed to RetrySettings.
+ *
+ * Example of creating a RetrySettings object using the constructor:
+ * ```
+ * $retrySettings = new RetrySettings([
+ *     'initialRetryDelayMillis' => 100,
+ *     'retryDelayMultiplier' => 1.3,
+ *     'maxRetryDelayMillis' => 60000,
+ *     'initialRpcTimeoutMillis' => 20000,
+ *     'rpcTimeoutMultiplier' => 1.0,
+ *     'maxRpcTimeoutMillis' => 20000,
+ *     'totalTimeoutMillis' => 600000,
+ *     'retryableCodes' => [ApiStatus::DEADLINE_EXCEEDED, ApiStatus::UNAVAILABLE],
```

Comment:
I know it exists in this form in other places (in Java too), but what is the correct way to spell this word (probably none, because it does not exist in dictionaries): "retryable" or "`retriable`"? At least when I try to translate both of these in google translate from English to Russian, only the second one gets translated.

Reply:
from [this source](https://english.stackexchange.com/questions/305273/retriable-or-retryable): 

> it's impossible to say that one spelling is correct and the other is incorrect

> Retryable is the more correct spelling, triable in the sense of able to be tested, is rare. Triable has a different meaning.

Dictionary.com does not have either one, while wiktionary only has a definition for [retriable](https://en.wiktionary.org/wiki/retriable). It looks like either one is appropriate, and the more important thing is consistency.

Reply:
The results are in! And very inconclusive, but more libraries use `retryable` than `retriable`

- Ruby uses just [retryable](https://github.com/googleapis/gax-ruby/search?utf8=%E2%9C%93&q=retryable&type=)
- Python uses just [retryable](https://github.com/googleapis/gax-python/search?utf8=%E2%9C%93&q=retryable&type=)
- NodeJS uses just [retryable](https://github.com/googleapis/gax-nodejs/search?utf8=%E2%9C%93&q=retryable&type=)
- Go uses just [retriable](https://github.com/googleapis/gax-go/search?utf8=%E2%9C%93&q=retriable&type=)
- Java uses both [retryable](https://github.com/googleapis/gax-java/search?utf8=%E2%9C%93&q=retryble&type=) and [retriable](https://github.com/googleapis/gax-java/search?utf8=%E2%9C%93&q=retriable&type=)
- DotNet uses neither

Reply:
I will leave as-is for now.